lib/platform/bitbucket/index.js

```diff
@@ -0,0 +1,596 @@
+  // Comments
+  ensureComment,
+  ensureCommentRemoval,
+  // PR
+  getPrList,
+  findPr,
+  createPr,
+  getPr,
+  getPrFiles,
+  updatePr,
+  mergePr,
+  // file
+  commitFilesToBranch,
+  getFile,
+  // Commits
+  getCommitMessages,
+};
+
+// Get all repositories that the user has access to
+async function getRepos(token, endpoint) {
+  logger.debug('getRepos(token, endpoint)');
+  const opts = endpoints.find({ platform: 'bitbucket' }, { token, endpoint });
+  if (!opts.token) {
+    throw new Error('No token found for getRepos');
+  }
+  endpoints.update({ ...opts, platform: 'bitbucket', default: true });
+  try {
+    const repos = await utils.accumulateValues(
+      `/2.0/repositories/?role=contributor`
+    );
+    return repos.map(repo => repo.full_name);
+  } catch (err) /* istanbul ignore next */ {
+    logger.error({ err }, `bitbucket getRepos error`);
+    throw err;
+  }
+}
+
+// Initialize bitbucket by getting base branch and SHA
+async function initRepo({ repository, token, endpoint }) {
+  logger.debug(`initRepo("${repository}")`);
+  const opts = endpoints.find({ platform: 'bitbucket' }, { token, endpoint });
+  if (!opts.token) {
+    throw new Error(`No token found for Bitbucket repository ${repository}`);
+  }
+  endpoints.update({ ...opts, platform: 'bitbucket', default: true });
+  api.reset();
+  config = {};
+  // TODO: get in touch with @rarkins about lifting up the caching into the app layer
+  config.repository = repository;
+  const platformConfig = {};
+  try {
+    const info = utils.repoInfoTransformer(
+      (await api.get(`/2.0/repositories/${repository}`)).body
+    );
+    platformConfig.privateRepo = info.privateRepo;
+    platformConfig.isFork = info.isFork;
+    platformConfig.repoFullName = info.repoFullName;
+    config.owner = info.owner;
+    logger.debug(`${repository} owner = ${config.owner}`);
+    config.defaultBranch = info.mainbranch;
+    config.baseBranch = config.defaultBranch;
+    config.mergeMethod = info.mergeMethod;
+  } catch (err) /* istanbul ignore next */ {
+    if (err.statusCode === 404) {
+      throw new Error('not-found');
+    }
+    logger.info({ err }, 'Unknown Bitbucket initRepo error');
+    throw err;
+  }
+  delete config.issueList;
+  delete config.prList;
+  delete config.fileList;
+  await Promise.all([getPrList(), getFileList()]);
+  return platformConfig;
+}
+
+// Returns true if repository has rule enforcing PRs are up-to-date with base branch before merging
+function getRepoForceRebase() {
+  // BB doesnt have an option to flag staled branches
+  return false;
+}
+
+async function setBaseBranch(branchName) {
+  if (branchName) {
+    logger.debug(`Setting baseBranch to ${branchName}`);
+    config.baseBranch = branchName;
+    delete config.baseCommitSHA;
+    delete config.fileList;
+    await getFileList(branchName);
+  }
+}
+
+// Search
+
+// Get full file list
+async function getFileList(branchName) {
+  if (config.fileList) {
+    return config.fileList;
+  }
```

Comment:
this looks wrong, since it does not check what the value of `branchName` is, could you confirm?

Reply:
Yes, it looks wrong. But I think I noticed the same problem in other platforms a while ago too. At the time I realised though that for the 99% use case, using the file list from the base branch is fine because the Renovate bot never *adds or deletes* files in its branches anyway. It has the potential to cause problems when multiple base branches are in use, but maybe that's actually working fine partly due to how/when it's called.